packages/dotcom-build-code-splitting/src/index.ts

```diff
@@ -1,4 +1,100 @@
-import { plugin } from './plugin'
+import assignDeep from 'assign-deep'
+import ReliableModuleIdsPlugin from 'reliable-module-ids-plugin'
+import type webpack from 'webpack'
 
-export { plugin }
-export default plugin
+import {
+  createBundleWithPackages,
+  createBundleWithRegExp,
+  createBundlesForPackages,
+  createBundlesForRegExp
+} from './bundleTypes'
+
+export class PageKitCodeSplittingPlugin {
+  apply(compiler: webpack.Compiler) {
+    const addInitialCodeSplitting = {
+      optimization: {
+        // Creates a separate bundle for webpack runtime.
+        // Specifying the name prevents multiple runtime bundles from being created.
+        runtimeChunk: {
+          name: 'webpack-runtime'
+        },
+        splitChunks: {
+          chunks: 'all'
+        },
+        // We're going to implement our own algorithm so don't double effort
+        moduleIds: false,
+        chunkIds: 'named'
+      }
+    }
```

Comment:
I think this could be simplified in future, the `cacheGroups` could be specified right here to avoid the need to deep merge everything later